frontend/src/components/teacher/session-creation/steps/AddInformation.tsx

```diff
@@ -1,22 +1,90 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
-import React from "react";
-import { Text } from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import {
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  HStack,
+  Textarea,
+  VStack,
+} from "@chakra-ui/react";
+
+import { isPastDate } from "../../../../utils/GeneralUtils";
+import DatePicker from "../../../common/DatePicker";
+import DistributeAssessmentWrapper from "../DistributeAssessmentWrapper";
 
 interface AddInformationProps {
+  startDate: Date | null;
+  endDate: Date | null;
+  notes: string;
   setStartDate: React.Dispatch<React.SetStateAction<Date | null>>;
   setEndDate: React.Dispatch<React.SetStateAction<Date | null>>;
   setNotes: React.Dispatch<React.SetStateAction<string>>;
+  setValidDates: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
 const AddInformation = ({
+  startDate,
+  endDate,
+  notes,
   setStartDate,
   setEndDate,
   setNotes,
+  setValidDates,
 }: AddInformationProps): React.ReactElement => {
+  const [invalidStartDate, setInvalidStartDate] = useState(false);
+  const [invalidEndDate, setInvalidEndDate] = useState(false);
+
+  useEffect(() => {
+    if (startDate) {
+      setInvalidStartDate(isPastDate(startDate));
+    }
+    if (endDate) {
+      setInvalidEndDate(isPastDate(endDate));
+    }
+    if (startDate && endDate) {
+      setInvalidStartDate(isPastDate(startDate) || endDate <= startDate);
+      setInvalidEndDate(isPastDate(endDate) || endDate <= startDate);
+    }
```

Comment:
```suggestion
    if (startDate && endDate) {
      setInvalidStartDate(isPastDate(startDate) || endDate <= startDate);
      setInvalidEndDate(isPastDate(endDate) || endDate <= startDate);
      return;
    } 

    if (startDate) {
      setInvalidStartDate(isPastDate(startDate));
    } 
    if (endDate) {
      setInvalidEndDate(isPastDate(endDate));
    }
```
reorder so we don't have to recompute?

Reply:
good catch